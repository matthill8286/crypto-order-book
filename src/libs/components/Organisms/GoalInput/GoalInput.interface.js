
//# sourceMappingURL=GoalInput.interface.js.map