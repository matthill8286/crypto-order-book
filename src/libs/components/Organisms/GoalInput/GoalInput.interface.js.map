{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"GoalInput.interface.js"}