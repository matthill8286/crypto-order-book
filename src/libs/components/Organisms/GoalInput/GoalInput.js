import React from 'react';
import { InputRange } from "../../Molecules/InputRange";
import { breakpoints, styled } from "../../../styles";
import { Heading, Typo } from "../../Atoms/Typography";
import { useWindowDimensions } from "../../Helper/useWindowDimensions";
var InputContainer = styled.div.withConfig({
  displayName: "GoalInput__InputContainer",
  componentId: "sc-5sx1y3-0"
})(["display:flex;flex-direction:column;width:95%;margin:0 auto;"]);
var RangeContainer = styled.div.withConfig({
  displayName: "GoalInput__RangeContainer",
  componentId: "sc-5sx1y3-1"
})(["display:inline-flex;flex:1;"]);
var StyledInputRange = styled(InputRange).withConfig({
  displayName: "GoalInput__StyledInputRange",
  componentId: "sc-5sx1y3-2"
})(["display:flex;flex-grow:1;"]);
var StyledTypo = styled(Typo).withConfig({
  displayName: "GoalInput__StyledTypo",
  componentId: "sc-5sx1y3-3"
})(["align-self:center;height:100%;"]);
var InformationContainer = styled.div.withConfig({
  displayName: "GoalInput__InformationContainer",
  componentId: "sc-5sx1y3-4"
})([""]);
export var GoalInput = function GoalInput(_ref) {
  var headline = _ref.headline,
      whereInputLabel = _ref.whereInputLabel,
      needInputLabel = _ref.needInputLabel,
      leftInputLabel = _ref.leftInputLabel,
      rightInputLabel = _ref.rightInputLabel,
      input = _ref.input,
      onProficiencyChange = _ref.onProficiencyChange,
      onImportanceChange = _ref.onImportanceChange,
      whereInputName = _ref.whereInputName,
      needInputName = _ref.needInputName,
      _ref$withBubble = _ref.withBubble,
      withBubble = _ref$withBubble === void 0 ? false : _ref$withBubble,
      _ref$showMarks = _ref.showMarks,
      showMarks = _ref$showMarks === void 0 ? false : _ref$showMarks;

  var _useWindowDimensions = useWindowDimensions(),
      currentBreakpoint = _useWindowDimensions.breakpoint;

  var isMobile = currentBreakpoint < breakpoints.lg;
  return React.createElement(InputContainer, null, React.createElement(InformationContainer, null, !isMobile && React.createElement(Heading, {
    scale: 'level-4',
    margin: 'xs 0',
    bold: true
  }, headline), React.createElement(Typo, {
    tag: 'div',
    color: 'grey4',
    fontSize: 'sm',
    padding: 'xs 0'
  }, whereInputLabel)), React.createElement(RangeContainer, null, React.createElement(StyledTypo, {
    tag: 'div',
    display: 'flex',
    color: 'grey4'
  }, leftInputLabel), React.createElement(StyledInputRange, {
    min: 0,
    max: 1,
    step: 0.1,
    markAmount: 1,
    name: whereInputName,
    onChange: onProficiencyChange,
    value: input.proficiency,
    showMarks: showMarks,
    withBubble: withBubble
  }), React.createElement(StyledTypo, {
    tag: 'div',
    display: 'flex',
    color: 'grey4'
  }, rightInputLabel)), React.createElement(InformationContainer, null, React.createElement(Typo, {
    tag: 'div',
    color: 'grey4',
    fontSize: 'sm',
    padding: 'xs 0'
  }, needInputLabel)), React.createElement(RangeContainer, null, React.createElement(StyledTypo, {
    tag: 'span',
    display: 'flex',
    color: 'grey4'
  }, leftInputLabel), React.createElement(StyledInputRange, {
    min: 0,
    max: 1,
    step: 0.1,
    markAmount: 1,
    name: needInputName,
    onChange: onImportanceChange,
    value: input.importance,
    showMarks: showMarks,
    withBubble: withBubble
  }), React.createElement(StyledTypo, {
    tag: 'span',
    display: 'flex',
    color: 'grey4'
  }, rightInputLabel)));
};
//# sourceMappingURL=GoalInput.js.map