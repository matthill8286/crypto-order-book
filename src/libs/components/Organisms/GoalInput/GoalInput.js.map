{"version":3,"sources":["../../../../src/components/Organisms/GoalInput/GoalInput.tsx"],"names":["React","InputRange","breakpoints","styled","Heading","Typo","useWindowDimensions","InputContainer","div","RangeContainer","StyledInputRange","StyledTypo","InformationContainer","GoalInput","headline","whereInputLabel","needInputLabel","leftInputLabel","rightInputLabel","input","onProficiencyChange","onImportanceChange","whereInputName","needInputName","withBubble","showMarks","currentBreakpoint","breakpoint","isMobile","lg","proficiency","importance"],"mappings":"AAAA,OAAOA,KAAP,MAAgC,OAAhC;AACA,SAASC,UAAT;AACA,SAASC,WAAT,EAAkCC,MAAlC;AACA,SAASC,OAAT,EAAkBC,IAAlB;AACA,SAASC,mBAAT;AAGA,IAAMC,cAAc,GAAGJ,MAAM,CAACK,GAAV;AAAA;AAAA;AAAA,mEAApB;AAMA,IAAMC,cAAc,GAAGN,MAAM,CAACK,GAAV;AAAA;AAAA;AAAA,mCAApB;AAKA,IAAME,gBAAgB,GAAGP,MAAM,CAACF,UAAD,CAAT;AAAA;AAAA;AAAA,iCAAtB;AAKA,IAAMU,UAAU,GAAGR,MAAM,CAACE,IAAD,CAAT;AAAA;AAAA;AAAA,sCAAhB;AAKA,IAAMO,oBAAoB,GAAGT,MAAM,CAACK,GAAV;AAAA;AAAA;AAAA,QAA1B;AAEA,OAAO,IAAMK,SAAmC,GAAG,SAAtCA,SAAsC,OAa7C;AAAA,MAZJC,QAYI,QAZJA,QAYI;AAAA,MAXJC,eAWI,QAXJA,eAWI;AAAA,MAVJC,cAUI,QAVJA,cAUI;AAAA,MATJC,cASI,QATJA,cASI;AAAA,MARJC,eAQI,QARJA,eAQI;AAAA,MAPJC,KAOI,QAPJA,KAOI;AAAA,MANJC,mBAMI,QANJA,mBAMI;AAAA,MALJC,kBAKI,QALJA,kBAKI;AAAA,MAJJC,cAII,QAJJA,cAII;AAAA,MAHJC,aAGI,QAHJA,aAGI;AAAA,6BAFJC,UAEI;AAAA,MAFJA,UAEI,gCAFS,KAET;AAAA,4BADJC,SACI;AAAA,MADJA,SACI,+BADQ,KACR;;AAAA,6BACsCnB,mBAAmB,EADzD;AAAA,MACgBoB,iBADhB,wBACIC,UADJ;;AAEJ,MAAMC,QAAQ,GAAGF,iBAAiB,GAAGxB,WAAW,CAAC2B,EAAjD;AACA,SACE,oBAAC,cAAD,QACE,oBAAC,oBAAD,QACG,CAACD,QAAD,IACC,oBAAC,OAAD;AAAS,IAAA,KAAK,EAAE,SAAhB;AAA2B,IAAA,MAAM,EAAE,MAAnC;AAA2C,IAAA,IAAI;AAA/C,KACGd,QADH,CAFJ,EAME,oBAAC,IAAD;AAAM,IAAA,GAAG,EAAE,KAAX;AAAkB,IAAA,KAAK,EAAE,OAAzB;AAAkC,IAAA,QAAQ,EAAE,IAA5C;AAAkD,IAAA,OAAO,EAAE;AAA3D,KACGC,eADH,CANF,CADF,EAWE,oBAAC,cAAD,QACE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,KAAjB;AAAwB,IAAA,OAAO,EAAE,MAAjC;AAAyC,IAAA,KAAK,EAAE;AAAhD,KACGE,cADH,CADF,EAIE,oBAAC,gBAAD;AACE,IAAA,GAAG,EAAE,CADP;AAEE,IAAA,GAAG,EAAE,CAFP;AAGE,IAAA,IAAI,EAAE,GAHR;AAIE,IAAA,UAAU,EAAE,CAJd;AAKE,IAAA,IAAI,EAAEK,cALR;AAME,IAAA,QAAQ,EAAEF,mBANZ;AAOE,IAAA,KAAK,EAAED,KAAK,CAACW,WAPf;AAQE,IAAA,SAAS,EAAEL,SARb;AASE,IAAA,UAAU,EAAED;AATd,IAJF,EAeE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,KAAjB;AAAwB,IAAA,OAAO,EAAE,MAAjC;AAAyC,IAAA,KAAK,EAAE;AAAhD,KACGN,eADH,CAfF,CAXF,EA8BE,oBAAC,oBAAD,QACE,oBAAC,IAAD;AAAM,IAAA,GAAG,EAAE,KAAX;AAAkB,IAAA,KAAK,EAAE,OAAzB;AAAkC,IAAA,QAAQ,EAAE,IAA5C;AAAkD,IAAA,OAAO,EAAE;AAA3D,KACGF,cADH,CADF,CA9BF,EAmCE,oBAAC,cAAD,QACE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,MAAjB;AAAyB,IAAA,OAAO,EAAE,MAAlC;AAA0C,IAAA,KAAK,EAAE;AAAjD,KACGC,cADH,CADF,EAIE,oBAAC,gBAAD;AACE,IAAA,GAAG,EAAE,CADP;AAEE,IAAA,GAAG,EAAE,CAFP;AAGE,IAAA,IAAI,EAAE,GAHR;AAIE,IAAA,UAAU,EAAE,CAJd;AAKE,IAAA,IAAI,EAAEM,aALR;AAME,IAAA,QAAQ,EAAEF,kBANZ;AAOE,IAAA,KAAK,EAAEF,KAAK,CAACY,UAPf;AAQE,IAAA,SAAS,EAAEN,SARb;AASE,IAAA,UAAU,EAAED;AATd,IAJF,EAeE,oBAAC,UAAD;AAAY,IAAA,GAAG,EAAE,MAAjB;AAAyB,IAAA,OAAO,EAAE,MAAlC;AAA0C,IAAA,KAAK,EAAE;AAAjD,KACGN,eADH,CAfF,CAnCF,CADF;AAyDD,CAzEM","sourcesContent":["import React, { useState } from 'react'\nimport { InputRange } from '@/components/Molecules/InputRange'\nimport { breakpoints, css, media, styled } from '@/styles'\nimport { Heading, Typo } from '@/components/Atoms/Typography'\nimport { useWindowDimensions } from '@/components/Helper/useWindowDimensions'\nimport { GoalInputProps } from '@/components/Organisms/GoalInput/GoalInput.interface'\n\nconst InputContainer = styled.div`\n  display: flex;\n  flex-direction: column;\n  width: 95%;\n  margin: 0 auto;\n`\nconst RangeContainer = styled.div`\n  display: inline-flex;\n  flex: 1;\n`\n\nconst StyledInputRange = styled(InputRange)`\n  display: flex;\n  flex-grow: 1;\n`\n\nconst StyledTypo = styled(Typo)`\n  align-self: center;\n  height: 100%;\n`\n\nconst InformationContainer = styled.div``\n\nexport const GoalInput: React.FC<GoalInputProps> = ({\n  headline,\n  whereInputLabel,\n  needInputLabel,\n  leftInputLabel,\n  rightInputLabel,\n  input,\n  onProficiencyChange,\n  onImportanceChange,\n  whereInputName,\n  needInputName,\n  withBubble = false,\n  showMarks = false,\n}) => {\n  const { breakpoint: currentBreakpoint } = useWindowDimensions()\n  const isMobile = currentBreakpoint < breakpoints.lg\n  return (\n    <InputContainer>\n      <InformationContainer>\n        {!isMobile && (\n          <Heading scale={'level-4'} margin={'xs 0'} bold>\n            {headline}\n          </Heading>\n        )}\n        <Typo tag={'div'} color={'grey4'} fontSize={'sm'} padding={'xs 0'}>\n          {whereInputLabel}\n        </Typo>\n      </InformationContainer>\n      <RangeContainer>\n        <StyledTypo tag={'div'} display={'flex'} color={'grey4'}>\n          {leftInputLabel}\n        </StyledTypo>\n        <StyledInputRange\n          min={0}\n          max={1}\n          step={0.1}\n          markAmount={1}\n          name={whereInputName}\n          onChange={onProficiencyChange}\n          value={input.proficiency}\n          showMarks={showMarks}\n          withBubble={withBubble}\n        />\n        <StyledTypo tag={'div'} display={'flex'} color={'grey4'}>\n          {rightInputLabel}\n        </StyledTypo>\n      </RangeContainer>\n      <InformationContainer>\n        <Typo tag={'div'} color={'grey4'} fontSize={'sm'} padding={'xs 0'}>\n          {needInputLabel}\n        </Typo>\n      </InformationContainer>\n      <RangeContainer>\n        <StyledTypo tag={'span'} display={'flex'} color={'grey4'}>\n          {leftInputLabel}\n        </StyledTypo>\n        <StyledInputRange\n          min={0}\n          max={1}\n          step={0.1}\n          markAmount={1}\n          name={needInputName}\n          onChange={onImportanceChange}\n          value={input.importance}\n          showMarks={showMarks}\n          withBubble={withBubble}\n        />\n        <StyledTypo tag={'span'} display={'flex'} color={'grey4'}>\n          {rightInputLabel}\n        </StyledTypo>\n      </RangeContainer>\n    </InputContainer>\n  )\n}\n"],"file":"GoalInput.js"}