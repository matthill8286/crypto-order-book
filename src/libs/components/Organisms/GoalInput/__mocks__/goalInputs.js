export var goalInputMocks = [{
  id: 1,
  proficiency: 0.7,
  importance: 0.1,
  competencyName: 'Making the most of Microsoft Teams'
}, {
  id: 32,
  proficiency: 0.5,
  importance: 0.3,
  competencyName: 'Managing a virtual team'
}, {
  id: 5,
  proficiency: 0.2,
  importance: 0.2,
  competencyName: 'New Technologies'
}];
//# sourceMappingURL=goalInputs.js.map