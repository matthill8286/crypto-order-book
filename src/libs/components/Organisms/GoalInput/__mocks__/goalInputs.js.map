{"version":3,"sources":["../../../../../src/components/Organisms/GoalInput/__mocks__/goalInputs.ts"],"names":["goalInputMocks","id","proficiency","importance","competencyName"],"mappings":"AAEA,OAAO,IAAMA,cAA8B,GAAG,CAC5C;AACEC,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,WAAW,EAAE,GAFf;AAGEC,EAAAA,UAAU,EAAE,GAHd;AAIEC,EAAAA,cAAc,EAAE;AAJlB,CAD4C,EAO5C;AACEH,EAAAA,EAAE,EAAE,EADN;AAEEC,EAAAA,WAAW,EAAE,GAFf;AAGEC,EAAAA,UAAU,EAAE,GAHd;AAIEC,EAAAA,cAAc,EAAE;AAJlB,CAP4C,EAa5C;AACEH,EAAAA,EAAE,EAAE,CADN;AAEEC,EAAAA,WAAW,EAAE,GAFf;AAGEC,EAAAA,UAAU,EAAE,GAHd;AAIEC,EAAAA,cAAc,EAAE;AAJlB,CAb4C,CAAvC","sourcesContent":["import { GoalDataProp } from '@/components/Organisms/GoalInput'\n\nexport const goalInputMocks: GoalDataProp[] = [\n  {\n    id: 1,\n    proficiency: 0.7,\n    importance: 0.1,\n    competencyName: 'Making the most of Microsoft Teams',\n  },\n  {\n    id: 32,\n    proficiency: 0.5,\n    importance: 0.3,\n    competencyName: 'Managing a virtual team',\n  },\n  {\n    id: 5,\n    proficiency: 0.2,\n    importance: 0.2,\n    competencyName: 'New Technologies',\n  },\n]\n"],"file":"goalInputs.js"}