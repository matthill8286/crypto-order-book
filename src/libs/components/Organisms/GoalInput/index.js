export * from "./GoalInput";
export * from "./GoalInput.interface";
//# sourceMappingURL=index.js.map