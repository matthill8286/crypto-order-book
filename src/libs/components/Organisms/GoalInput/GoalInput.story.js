function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { withState } from '@dump247/storybook-state';
import { boolean, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { GoalInput } from "./GoalInput";
import { goalInputMocks } from "./__mocks__/goalInputs";
var stories = storiesOf('Design System/Organisms/GoalInput', module);
var initialState = goalInputMocks[0];
stories.add('Default', withState({
  proficiency: 0.2,
  importance: 0.4
}, function (store) {
  var onProficiencyChange = function onProficiencyChange(event) {
    var value = event.target.value;
    store.set({
      proficiency: value
    });
  };

  var onImportanceChange = function onImportanceChange(event) {
    var value = event.target.value;
    store.set({
      importance: value
    });
  };

  var knobs = function knobs() {
    return {
      headline: text('Headline', 'Making the most of Microsoft Teams'),
      whereInputLabel: text('Where Label', 'Where you are now:'),
      needInputLabel: text('Need Label', 'Where you need to be:'),
      whereInputName: text('Name for Where', 'where'),
      needInputName: text('name for Need', 'need'),
      withBubble: boolean('With Bubble', false),
      showMarks: boolean('Show Marks', false)
    };
  };

  return React.createElement(GoalInput, _extends({
    input: {
      proficiency: store.state.proficiency,
      id: 4,
      importance: store.state.importance,
      competencyName: 'Making the most of Microsoft Teams'
    },
    leftInputLabel: 'Beginner',
    rightInputLabel: 'Master',
    onProficiencyChange: onProficiencyChange,
    onImportanceChange: onImportanceChange
  }, knobs(), {
    whereInputName: 'where-input',
    needInputName: 'need-input'
  }));
}));
//# sourceMappingURL=GoalInput.story.js.map