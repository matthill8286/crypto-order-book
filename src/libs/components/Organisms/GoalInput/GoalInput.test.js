import React from 'react';
import { mountWithTheme, renderWithTheme } from "../../../testRenderer";
import { GoalInput } from "./GoalInput";
import { goalInputMocks } from "./__mocks__/goalInputs";
describe('GoalInput', function () {
  it('should render', function () {
    var onChange = jest.fn();
    var wrapper = renderWithTheme(React.createElement(GoalInput, {
      whereInputName: 'test-goal',
      needInputName: 'test-need',
      needInputLabel: 'Where do you need to be:',
      headline: 'Can I manage you as well as I should?',
      whereInputLabel: 'Where are you now:',
      input: goalInputMocks[0],
      onProficiencyChange: onChange,
      onImportanceChange: onChange
    }));
    expect(wrapper).toMatchSnapshot();
  });
  it('should fire change event', function () {
    var onChange = jest.fn();
    var wrapper = mountWithTheme(React.createElement(GoalInput, {
      whereInputName: 'where-input',
      needInputName: 'need-input',
      needInputLabel: 'Where do you need to be:',
      headline: 'Can I manage you as well as I should?',
      whereInputLabel: 'Where are you now:',
      input: goalInputMocks[0],
      onProficiencyChange: onChange,
      onImportanceChange: onChange
    }));
    var needInput = wrapper.find('input[name*="need-input"]');
    var whereInput = wrapper.find('input[name*="where-input"]');
    needInput.simulate('change');
    expect(onChange).toBeCalledTimes(1);
    whereInput.simulate('change');
    expect(onChange).toBeCalledTimes(2);
  });
});
//# sourceMappingURL=GoalInput.test.js.map