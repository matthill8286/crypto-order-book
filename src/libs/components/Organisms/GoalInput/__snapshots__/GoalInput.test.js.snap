// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalInput should render 1`] = `
<div
  className="GoalInput__InputContainer-sc-5sx1y3-0 litotK"
>
  <div
    className="GoalInput__InformationContainer-sc-5sx1y3-4"
  >
    <h4
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hGcpCs"
      fontFamily="default"
      fontSize="lg"
      spacing="base"
    >
      Can I manage you as well as I should?
    </h4>
    <div
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 cpKuUF"
      color="grey4"
      fontSize="sm"
    >
      Where are you now:
    </div>
  </div>
  <div
    className="GoalInput__RangeContainer-sc-5sx1y3-1 jfkOHz"
  >
    <div
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWArgc GoalInput__StyledTypo-sc-5sx1y3-3 kgfwyf"
      color="grey4"
      display="flex"
    />
    <div
      className="InputRangestyled__StyledInputRangeContainer-sc-10ybtf3-0 kUxsRe"
    >
      <div
        className="InputRangestyled__StyledInputRangeThumb-sc-10ybtf3-5 jnrEVz"
        style={
          Object {
            "left": "70%",
          }
        }
      />
      <input
        className="InputRangestyled__StyledInputRange-sc-10ybtf3-6 bhLsAS"
        list="test-goal"
        max={1}
        min={0}
        name="test-goal"
        onChange={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step={0.1}
        type="range"
        value={0.7}
      />
    </div>
    <div
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWArgc GoalInput__StyledTypo-sc-5sx1y3-3 kgfwyf"
      color="grey4"
      display="flex"
    />
  </div>
  <div
    className="GoalInput__InformationContainer-sc-5sx1y3-4"
  >
    <div
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 cpKuUF"
      color="grey4"
      fontSize="sm"
    >
      Where do you need to be:
    </div>
  </div>
  <div
    className="GoalInput__RangeContainer-sc-5sx1y3-1 jfkOHz"
  >
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWArgc GoalInput__StyledTypo-sc-5sx1y3-3 kgfwyf"
      color="grey4"
      display="flex"
    />
    <div
      className="InputRangestyled__StyledInputRangeContainer-sc-10ybtf3-0 kUxsRe"
    >
      <div
        className="InputRangestyled__StyledInputRangeThumb-sc-10ybtf3-5 jnrEVz"
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <input
        className="InputRangestyled__StyledInputRange-sc-10ybtf3-6 bhLsAS"
        list="test-need"
        max={1}
        min={0}
        name="test-need"
        onChange={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step={0.1}
        type="range"
        value={0.1}
      />
    </div>
    <span
      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hWArgc GoalInput__StyledTypo-sc-5sx1y3-3 kgfwyf"
      color="grey4"
      display="flex"
    />
  </div>
</div>
`;
